### PR TITLE
Delay action

### PR DESCRIPTION
**Pull request by dannyroosevelt**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by dannyroosevelt** [comment]
Btw this was my prototype, so any refactoring / code changes, etc that you want to see, go for it -- whoever is reviewing, please just go ahead and implement whatever recommendations you have.

File: components/delay/actions/workflow-delay/workflow-delay.mjs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      propDefinition: [
+        delay,
+        "delayDurationUnit",
+      ],
+      label: "Duration to delay (unit)",
+      description: "Specify the time unit for delaying the workflow.",
+    },
+  },
+  type: "action",
+  methods: {},
```
Comment (jcortes) [inline components/delay/actions/workflow-delay/workflow-delay.mjs:28]: I think these two properties can be removed

File: components/delay/actions/workflow-delay/workflow-delay.mjs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  async run({ $ }) {
+    const milliseconds = this.delay.convertToMilliseconds(
+      this.delayDurationUnit,
+      this.delayDurationValue,
+    );
+    const resp = $.flow.delay(milliseconds);
```
Comment (jcortes) [inline components/delay/actions/workflow-delay/workflow-delay.mjs:34]: shouldn't this line be await?
```suggestion
    const resp = await $.flow.delay(milliseconds);
```

**Review by jcortes — CHANGES_REQUESTED** [review]
@dannyroosevelt  It looks good to me, just added a couple of comments and that's it

**Review by jcortes — APPROVED** [review]
@dannyroosevelt  LGTM!

**Pull request merged**